### PR TITLE
Fix price filtering

### PR DESCRIPTION
**Pull request by AndreLemay**
Adds a filter to the price column that uses an operator field with an amount field, looks like this:

image

Only the amount field actually triggers the filtering, and the trigger itself is debounced so it should only update the table when the user is finished entering the amount. We can probably use the same strategy to get the real-time search working again.

Note: I had to do some weird stuff with props/state on the filter component in order to get the amount field to work properly with the debouncing. I can explain in more detail if you want, as it might affect any other custom filters we want to add.

### CODE REVIEW

File: src/components/VehiclesColumnProvider.js [hunk 1]
```diff
@@ -33,11 +34,6 @@ export default function VehiclesColumnProvider({ children, currentBook }) {
       { label: "Silhouette", name: "silhouette" },
       { label: "Speed", name: "speed" },
       { label: "Handling", name: "handling" },
-      { label: "Armor", name: "armor" },
```
Comment (duffn) [inline src/components/VehiclesColumnProvider.js]: Looks like these columns were unintentionally removed.

File: src/components/StarshipsColumnProvider.js [hunk 2]
```diff
@@ -33,11 +34,7 @@ export default function StarshipsColumnProvider({ children, currentBook }) {
       { label: "Silhouette", name: "silhouette" },
       { label: "Speed", name: "speed" },
       { label: "Handling", name: "handling" },
-      { label: "Armor", name: "armor" },
```
Comment (duffn) [inline src/components/StarshipsColumnProvider.js]: Looks like these columns were unintentionally removed.

**Comment by AndreLemay** [comment]
@duffn I currently have no idea why the test is failing. It's specifically the one checking that the restricted column is inserted correctly (which it definitely is, or else none of the tables with it would work). For whatever reason jest is just not agreeing that the column is equal to itself. I copy pasted the column into the test file, so I know that code-wise they're an exact match, but I suspect that however jest compares objects is not working with that one for some reason (which is strange, since it has no issue with the generated id column).

I've also been unable to get vscode to play nice with jest. The debugger just runs into a syntax error as soon as it starts up and tries to connect to the node server running jest. Haven't had a chance to really dig into the why yet. If you'd rather not make the tests no longer required, I can just comment out the failing test for now.

**Comment by duffn** [comment]
I’ve turned that required check off for now.

File: src/components/VehiclesColumnProvider.js (hunk 1 above)
Comment (AndreLemay) [reply src/components/VehiclesColumnProvider.js]: Oohhh yeah, didn't notice that. Git failed miserably at trying to merge a few of these and basically just highlighted half the column object as the "change".

**Comment by duffn** [comment]
The restricted/not restricted looks like it's fixed.

One other thing I just noticed is that the filter does not change if you only change the price dropdown - you actually need to change the value in order to get it to change in the table.

- Add a price filter for >= 10.
- Change >= to <.
- The filter on the table does not change.

**Comment by AndreLemay** [comment]
> The restricted/not restricted looks like it's fixed.
> 
> One other thing I just noticed is that the filter does not change if you only change the price dropdown - you actually need to change the value in order to get it to change in the table.
> 
> * Add a price filter for >= 10.
> * Change >= to <.
> * The filter on the table does not change.

Hm.... yeah that was intentional, my thinking was that when you first open the filter, you'd set the comparison you want, and it wouldn't make sense to update the table until you've set the value, but the scenario you present is totally valid. I'll change that so they both update.

**Comment by duffn** [comment]
I merged the humanize PR first re are some conflicts here now. Can you also just comment out the offending failing test while your at it?

**Comment by duffn** [comment]
Sorry, I didn't notice this during the first review, but the price gets cleared if you select another filter.

Example:

- On weapons, set a price filter.
- Set a category filter.
- The price filter is zeroed out.

**Comment by AndreLemay** [comment]
> Sorry, I didn't notice this during the first review, but the price gets cleared if you select another filter.
> 
> Example:
> 
> * On weapons, set a price filter.
> * Set a category filter.
> * The price filter is zeroed out.

I'm not seeing this behaviour on my local build or on the gatsby cloud deployment.

image

**Comment by duffn** [comment]
That's odd. I can't either now. Perhaps I was on a different branch locally? I thought I was on the preview build, but I can't repro this now either.

**Pull request merged**